options.go

```diff
@@ -5,22 +5,20 @@ package libp2p
 
 import (
 	"fmt"
-	"net"
-
 	"github.com/libp2p/go-libp2p-core/connmgr"
 	"github.com/libp2p/go-libp2p-core/crypto"
 	"github.com/libp2p/go-libp2p-core/metrics"
 	"github.com/libp2p/go-libp2p-core/network"
 	"github.com/libp2p/go-libp2p-core/peer"
 	"github.com/libp2p/go-libp2p-core/peerstore"
 	"github.com/libp2p/go-libp2p-core/pnet"
+	"github.com/pkg/errors"
```

Comment:
you don't need this dep. either the system built-in `"errors"` can be used for `errors.New`, or you can use the already included `fmt` for `fmt.Errorf()`.